### DISCUSSION TITLE
I can't make android AAB.

### DISCUSSION THREAD

**Discussion by kanisuki** [airsdk/Adobe-Runtime-Support#2536] [Q&A]
Hello there,
My tool are Win10, adobe animate Ver.23.0, Android Studio Ver.2022.1.1 patch2, AIR Ver. 50.2.1.1, 
In Android Studio welcome screen,
1. I chose "Profile or debug".
2. I chose my APK file.
3. I chose "Build".
But only "Generate Signed APK" appeared.
"Generate Signed Bundle/APK" was not appeared.
But if I made New Project , in case Basic Activity, "Generate Signed Bundle/APK" was appeared.
I might make mistake the order to make AAB?
Or I might make mistake to set something?

**Comment by ajwfrost** [comment]
To confirm this, it sounds like you are opening an APK file and then trying to create an AAB from this? I don't think that would work..

How did you get the APK file? If that was created via Animate, then there is a way to force it to create an AAB file instead. Unfortunately they haven't added the option on the user interface, but if you create a file:
`c:\users\username\.airsdk\adt.cfg`
and add to it the line:
```
CreateAndroidAppBundle=true
```
then when you publish from Animate again, it may tell you that it's an APK file (and it may well have an .apk extension) but it should be an Android App Bundle. (Worth opening it in a zip file browser to check - look for "BundleConfig.pb" at the top level).

thanks

**Reply by kanisuki** [reply]
Thank you, Sir.
I make APK file with  adobe Animate.
And I will make C :\ users\testapk\.airsdk\adt.cfg
And in this cfg, I will write "CreateAndroidAppBundle=true"
Is this OK?
I can't understand that  (Worth opening it in a zip file browser to check - look for "BundleConfig.pb" at the top level).

**Reply by ajwfrost** [reply]
Just to confirm, the config filename is "adt.cfg" and should be in a folder:
`c:\users\username\.airsdk`
i.e. `.airsdk` underneath `%userprofile%`

Once you have the file that's generated, it will still be called "projectname.apk" I think .. if you can open this up in a zip file manager (WinZip, 7-Zip, etc) to see what the file contents are (both APK and AAB files are actually just zips) then you can check whether it's a bundle or an APK file, by seeing if it has that "BundleConfig.pb" file in it...

thanks

**Reply by kokevk** [reply]
AirSdk/lib/adt.cfg >> CreateAndroidAppBundle=true

publish      .apk file
Rename     .apk file to .aab 
 it's that simple

**Reply by kanisuki** [reply]
Thak you , ajwfrost and kokevk. I will try....

**Reply by kanisuki** [reply]
Hello there,
I made AirSdk/lib/adt.cfg.  And write in it CreateAndroidAppBundle = true
Then I published my fla to AIR 50.2.1.1 for Android.
It made  some files contained .apk file.
Only rename this .apk to .aab?
I need not to use Android Studio?

I also use WinZip to look in this .apk,  There are some folders and files in it. What do I do?
I would like to express my gratitude for your helps.

**Reply by teotigraphix** [reply]
If your unzipped folder structure looks like the following, you have an `.aab` archive ready for upload to Google Play.

![image](https://user-images.githubusercontent.com/226417/227771440-b9b7c5d4-9df1-468e-bbc3-53352a2d576f.png)

Rename the file extension to `.aab` from `.apk` and then upload.

**Reply by kanisuki** [reply]
Hello kokevk, thank you,  Which ?
1.
c:\AirSdk\lib\adt.cfg
CreateAndroidAppBundle = true

2.
c:\AirSdk\lib\adt.cfg
AndroidPlatformSDK = C:\Users\\user\AppData\\Local\Android\sdk
JAVA_HOME = C:\AndroidStudio\jre

Which way Do I choose?

**Reply by kanisuki** [reply]
Hello teotigraphix, Thank you. 
my win zip scene is differ from this, has assets, lib and META-INF folders,Androidmainfest.xml. classes.dex ,resouces.arsc failes . 
I don't know how to uploard pictures.

**Reply by ajwfrost** [reply]
If you can see the AndroidManifest.xml file at the top level then it's an APK not an AAB... so, it seems like it's not picking up the `CreateAndroidAppBundle=true` line.

Are you able to use the AIR SDK Manager and point this at your `c:\airsdk` folder? it should then help with getting the settings sorted out properly and can also be used to trouble-shoot whether it's picking up the bundle override request..

thanks

**Reply by kanisuki** [reply]
Thank you ajwfrost,
I can't download AIR SDK Manager #2291. The link is not alive.

**Reply by marchbold** [reply]
Try the airsdk dev site:

- macos: https://airsdk.dev/docs/basics/install/macos
- windows: https://airsdk.dev/docs/basics/install/windows

**Reply by kanisuki** [reply]
Thank you marchbold, I am referring ....

**Comment by kokevk** [comment]
you don't need to use android studio
only this in the adt.cfg

`
AndroidPlatformSDK=C:\\Users\\user\\AppData\\Local\\Android\\sdk`

`JAVA_HOME=D:\\AndroidStudio\\jre`

you publish it in adobe animate and rename
